exonum-java-binding-cryptocurrency-demo/src/test/java/com/exonum/binding/cryptocurrency/transactions/TransferTxTest.java

```diff
@@ -142,69 +179,66 @@ void executeTransfer() throws CloseFailuresException {
 
   @Test
   @RequiresNativeLibrary
-  void executeTransferToTheSameWallet() throws CloseFailuresException {
+  void executeTransfer_NoSuchFromWallet() throws CloseFailuresException {
     try (Database db = MemoryDb.newInstance();
          Cleaner cleaner = new Cleaner()) {
       Fork view = db.createFork(cleaner);
+      // Create a receiver’s wallet with the given initial balance
+      long initialBalance = 50L;
+      createWallet(view, toKey, initialBalance);
 
-      long initialBalance = 100L;
-      createWallet(view, fromKey, initialBalance);
-
-      // Create and execute the transaction
       long seed = 1L;
-      long transferSum = 40L;
-      TransferTx tx = withMockMessage(seed, fromKey, fromKey, transferSum);
-      tx.execute(view);
 
-      // Check that the balance of the wallet remains the same
-      CryptocurrencySchema schema = new CryptocurrencySchema(view);
-      ProofMapIndexProxy<PublicKey, Wallet> wallets = schema.wallets();
-      assertThat(wallets.get(fromKey).getBalance(), equalTo(initialBalance));
+      long transferValue = 50L;
+      TransferTx tx = withMockMessage(seed, fromKey, toKey, transferValue);
+      // Execute the transaction that attempts to transfer from an unknown wallet
+      TransactionExecutionException e = assertThrows(
+          TransactionExecutionException.class, () -> tx.execute(view));
+      assertThat(e.getErrorCode(), equalTo(UNKNOWN_SENDER.errorCode));
```

Comment:
just for your taste 🐱 
```
private static void assertExecutionException(TransactionExecutionException e, TransactionError expected) {
    assertThat(e.getErrorCode(), equalTo(expected.errorCode))
}
```

Reply:
:grin: Thanks! I pulled that in a custom Hamcrest matcher, so that the readability of client code is better.

Reply:
the matcher looks very nice. It would be great to implement matchers for  exonum collections like `MapIndex` etc

Reply:
Hm, I think we might sometime put them in the upcoming testkit :+1: